src/main/java/uk/gov/hmcts/darts/addaudio/AddAudioRoute.java

```diff
@@ -61,57 +61,56 @@ public DARTSResponse route(AddAudio addAudio) {
-                    audiosClient.addAudioMetaData(DataUtil.convertToStorageGuid(metaData, blobStoreUuid.get()));
-                });
-            } else {
+            if (request.isEmpty()) {
                 log.error("The add audio endpoint requires a file to be specified. No file was found");
                 throw new DartsException(CodeAndMessage.ERROR);
             }
-        } catch (IOException ioe) {
-            throw new DartsException(ioe, CodeAndMessage.ERROR);
+            addAudioValidator.validateCourtroom(addAudioLegacy);
+
+            AtomicReference<String> checksum = new AtomicReference<>(null);
+            request.get().consumeFileBinaryStream(uploadedStream -> checksum.set(FileContentChecksum.calculate(uploadedStream.getInputStream())));
+            // consume the uploaded file and proxy downstream
+            request.get().consumeFileBinaryStream(uploadedStream -> {
+                StreamingMultipart multipartFile = new StreamingMultipart(
+                    addAudioLegacy.getMediafile(),
+                    addAudioLegacy.getMediaformat(),
+                    uploadedStream
+                );
+                multipartFileValidator.validate(multipartFile);
+
+                AddAudioMetadataRequest metaData = addAudioMapper.mapToDartsApi(addAudioLegacy);
+                addAudioValidator.validateSize(metaData, request.get().getBinarySize());
+
+                metaData.setFileSize(request.get().getBinarySize());
+                metaData.setChecksum(checksum.get());
+                try {
+                    blobStoreUuid.set(dataManagementService.saveBlobData(
+                        dataManagementConfiguration.getInboundContainerName(),
+                        multipartFile.getInputStream(),
+                        DataUtil.toMap(metaData)));
+                } catch (Exception e) {
+                    log.error("Failed to upload audio file to the inbound blob store", e);
+                    logApi.failedToLinkAudioToCases(
+                        metaData.getCourthouse(),
+                        metaData.getCourtroom(),
+                        metaData.getStartedAt(),
+                        metaData.getEndedAt(),
+                        metaData.getCases(),
+                        metaData.getChecksum(),
+                        null
+                    );
+                    throw new DartsException(e, CodeAndMessage.ERROR);
+                }
+                log.info("Audio file uploaded successfully to the inbound blob store. BlobStoreUuid: {}", blobStoreUuid);
+                audiosClient.addAudioMetaData(DataUtil.convertToStorageGuid(metaData, blobStoreUuid.get()));
+            });
+        } catch (DartsValidationException | ClientProblemException e) {
+            throw e;
         } catch (Exception e) {
             if (blobStoreUuid.get() != null) {
                 dataManagementService.deleteBlobData(dataManagementConfiguration.getInboundContainerName(), blobStoreUuid.get());
```

Comment:
For completeness, do you think we should add a log entry before the delete from blob store call so we have some sort of traceability around what was deleted?